lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -324,16 +327,46 @@ def _on_update_status(self, event: UpdateStatusEvent):
         # handle when/if certificates are expired
         self._check_certs_expiration(event)
 
-    def _on_get_admin_secrets_action(self, event: ActionEvent):
+    def _on_set_password_action(self, event: ActionEvent):
+        """Set new admin password from user input or generate if not passed."""
+        if not self.unit.is_leader():
+            event.fail("The action can be run only on leader unit.")
+            return
+
+        user_name = event.params.get("username")
+        if event.params.get("username") != "admin":
+            event.fail("Only the 'admin' username is allowed for this action.")
```

Comment:
**question** I might miss something here, but what is the point of making an action with a parameter username if I cannot really provide other values than admin?

Reply:
That is a good point, and it doesn't make much sense in OpenSearch at this stage at least. But the idea is to stay consistent with the UX of the other charms.